src/overlays/gamestates/ovl_debug_opening/z_debug_opening.c

```diff
@@ -0,0 +1,316 @@
+        Sram_InitDebugSave();
+    } else {
+        Sram_InitNewSave();
+    }
+    // Set age, time and entrance
+    gSaveContext.save.linkAge = BOOT_AGE;
+    gSaveContext.save.dayTime = CLOCK_TIME(12, 0);
+    gSaveContext.save.entranceIndex = BOOT_ENTRANCE;
+
+    gSaveContext.respawnFlag = 0;
+    gSaveContext.respawn[RESPAWN_MODE_DOWN].entranceIndex = ENTR_LOAD_OPENING;
+
+    // Have the magic meter load correctly
+    gSaveContext.magicFillTarget = gSaveContext.save.info.playerData.magic;
+    gSaveContext.magicCapacity = 0;
+    gSaveContext.save.info.playerData.magic = 0;
+    gSaveContext.save.info.playerData.magicLevel = 0;
+
+    gSaveContext.seqId = (u8)NA_BGM_DISABLED;
+
+    // Load Play state
+    gSaveContext.gameMode = GAMEMODE_NORMAL;
+    this->state.running = false;
+    SET_NEXT_GAMESTATE(&this->state, Play_Init, PlayState);
+}
+
+void DebugOpening_LoadTitleScreen(GameState* thisx) {
+    DebugOpeningState* this = (DebugOpeningState*)thisx;
+
+    gSaveContext.seqId = (u8)NA_BGM_DISABLED;
+    gSaveContext.natureAmbienceId = 0xFF;
+    gSaveContext.gameMode = GAMEMODE_TITLE_SCREEN;
+    this->state.running = false;
+    SET_NEXT_GAMESTATE(&this->state, TitleSetup_Init, TitleSetupState);
+}
+
+void DebugOpening_LoadFileSelect(GameState* thisx) {
+    DebugOpeningState* this = (DebugOpeningState*)thisx;
+
+    gSaveContext.gameMode = GAMEMODE_FILE_SELECT;
+    SET_NEXT_GAMESTATE(&this->state, FileSelect_Init, FileSelectState);
+    this->state.running = false;
+}
+
+void DebugOpening_LoadMapSelect(GameState* thisx) {
+    DebugOpeningState* this = (DebugOpeningState*)thisx;
+
+    gSaveContext.gameMode = GAMEMODE_NORMAL;
+    SET_NEXT_GAMESTATE(&this->state, MapSelect_Init, MapSelectState);
+    this->state.running = false;
+}
+
+/**
+ * sOptionInfo contains every usable option in the first page.
+ * The .name of every option is printed in DebugOpening_DrawOptions
+ * through a loop and the .func gets executed based on this->currentOption
+ * in DebugOpening_ControlOptions
+ */
+static OptionInfo sOptionInfo[] = {
+    { .func = DebugOpening_ChooseSaveFile, .name = "Save File: " },
+    {
+        .func = DebugOpening_LoadDefinedScene,
+        .name = "Init Defined Scene",
+    },
+    {
+        .func = DebugOpening_LoadMapSelect,
+        .name = "Init Map Select",
+    },
+    {
+        .func = DebugOpening_LoadTitleScreen,
+        .name = "Init Title Screen",
+    },
+    {
+        .func = DebugOpening_LoadFileSelect,
+        .name = "Init File Select",
+    },
+};
+
+void DebugOpening_ControlOptions(GameState* thisx) {
+    DebugOpeningState* this = (DebugOpeningState*)thisx;
+
+    // if dpad-up is pressed, go up in the options page
+    if (CHECK_BTN_ANY(this->state.input[0].press.button, BTN_DUP) && this->currentOption > OPTION_CHOOSE_SAVE_FILE) {
+        this->currentOption--;
+        Audio_PlaySfxGeneral(NA_SE_IT_SWORD_IMPACT, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
```

Comment:
just a detail but I think it'd be nice if it was using the same sfx the map select is using when going to the next line